### DISCUSSION TITLE
Subscribing a value of an array with client_example.py does not work

### DISCUSSION THREAD

**Discussion by pechtus3000** [FreeOpcUa/opcua-asyncio#1374] [General]
Dear community,

I have got the following problem/question: I want to subscribe to data changes of a node. This node doesn't have one value as most of the servers normally have with f.e. 100 Nodes and each one of them has got one single value to read/write as client. In my case, there only exists one node but this node has got an array as value. When I run the client-example.py script to subscribe to the node, the whole array is printed out in the terminal once. If I then change one value inside the array with the help of UA Expert nothing happens in the python script.

Does anyone here know how to subscribe to an array, but maybe only the first three indexes f.e. ?

**Comment by bauermarkus** [comment]
*Subscribe*

I have a similar problem

I subscribed to a variable in S7-1500 and I only get an event once

**Reply by pechtus3000** [reply]
I don't know how to change that. I would only need to subscribe four values of the whole array, but it won't work. My actual workaround in the meantime is to pull the whole array with a simple while loop:

while True:
    await myvar.read_value()
    var1 = myvar[0]
    print(var1)


which reads the whole array, but defines the value with index 0 as var1 and prints it out

**Reply by schroeder-** [reply]
Never monitored a array variable, maybe the server expect a special filter? Did you try to provide a custom filter? There you can also specify index ranges, to only get the wanted variables.

**Reply by pechtus3000** [reply]
Thank you @schroeder- 

To be honest, I don't know how to program a custom filter for a node in opc ua. Do you know if there are any examples given?

**Reply by schroeder-** [reply]
This is realy tricky, you won't need a filter but https://reference.opcfoundation.org/Core/Part4/v105/docs/7.21, to do this you have to go realy low level and craft the total request our self and then it could be possible the server can't processes this or ignore the content.

The first question is does it work with UAExpert? Because if the same behaviour is seen in UAExpert, it is most of the time a server bug.

**Reply by hsk0815** [reply]
Hi, i have he same problem. I tested it with UaExpert and it works fine. 
I updated my opcua lib from an older version which worked with my opc server. I was able to monitor one or more varibles and i did not have to implement a filter.

**Reply by pechtus3000** [reply]
@schroeder- it works fine with ua expert, my problem is that the subscription with the opc ua client doesn't work properly: when starting the program, it's printing out the whole array. If I then change one value inside the array with the help of opc ua nothing happens, but in ua expert the changed value is shown immediately
